### DISCUSSION TITLE
TreeSelect seems to be some features, is this since v4?

### DISCUSSION THREAD

**Discussion by james-astra** [primefaces/community#2702] [PrimeVue] [answered]
The TreeSelect component appears to be missing some useful features for it to be used in a lot of applications.

What's missing?

**Programatically expandable keys**
There is no 'expandedKeys' prop available on the component and seemingly no way to pass it through to the child Tree component. This feature is really useful.
I know that there is an 'expandPath' method available but also this doesn't do anything.

**Propagation control**
In my use case, I would like to propagate selections downwards. This is the default behaviour (select all children when a node is selected) however, I don't want to propagate upwards, in the case of a node with one child, I need them to be able to be selected independently of each other.

I am using it for an 'organization selector' that controls what data from what organizations a user is seeing.

**Comment by james-astra** [comment]
Any update regarding this?

**Comment by jelleroorda** [comment, marked as answer]
This seems possible, at least with latest version: https://primevue.org/treeselect/#api.treeselect.props.expandedKeys

**Reply by james-astra** [reply]
Yes looks like the prop is back in action now :) Thanks

**Reply by simonmargesin** [reply]
Have you found a workaround for the missing propagateSelectionUp and propagateSelectionDown functionalities? I also need this similarly for organizations and would prefer not to overwrite the whole component for features that technically already exist

**Reply by james-astra** [reply]
> Have you found a workaround for the missing propagateSelectionUp and propagateSelectionDown functionalities? I also need this similarly for organizations and would prefer not to overwrite the whole component for features that technically already exist

No, Tree & TreeSelect are the only PrimeVue components that I have just written bespoke components for (they also don't handle large datasets very well either)

**Reply by simonmargesin** [reply]
Yea i really get that, TreeSelect has been the most frustrating thing about Primevue so far. Had to switch away from FormFields, because the TreeSelect in Checkbox-mode could not handle them at all and now im missing basic functionalities that are apparently present in the basetree and in primeng
